### DISCUSSION TITLE
How to generate a two-dimensional boundary

### DISCUSSION THREAD

**Discussion by amosaha** [idaholab/moose#19920] [Q&A Meshing] [answered]
Hello,
The meshes I generated with trelis cannot be used, so I tried to use the mesh generator that comes with moose.I have created a 3D model. If I want to specify a specific 2D plane, which function should I use？
                                                                                                                                       Thanks.

**Comment by GiudGiud** [comment, marked as answer]
Hello

Do you want to use this plane as a sideset in a 3D solve (for a boundary condition for example) or do you want to use this plane as a 2D mesh for the solve?
Either way, you'll want to work with one of those to define the plane
https://mooseframework.inl.gov/source/meshgenerators/ParsedGenerateSideset.html
or if the plane is on the outer boundary of the mesh:
https://mooseframework.inl.gov/source/meshgenerators/SideSetsAroundSubdomainGenerator.html

Guillaume
